packages/grant-explorer/src/features/common/__tests__/CartNotification.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { render, fireEvent, screen } from "@testing-library/react";
+import CartNotification, { ProjectView } from "../CartNotification";
+import { makeApprovedProjectData } from "../../../test-utils";
+import { vi } from "vitest"; // Adjust the path accordingly
+import DefaultLogoImage from "../../../assets/default_logo.png";
+
+const mockProject = makeApprovedProjectData();
+
+describe("<CartNotification />", () => {
+  it("renders when showCartNotification is true", () => {
+    render(
+      <CartNotification
+        showCartNotification={true}
+        setShowCartNotification={() => {}}
+        currentProjectAddedToCart={mockProject}
+      />
+    );
+    expect(screen.getByText("Project added to your cart")).toBeInTheDocument();
+  });
+
+  it("does not render when showCartNotification is false", () => {
+    render(
+      <CartNotification
+        showCartNotification={false}
+        setShowCartNotification={() => {}}
+        currentProjectAddedToCart={mockProject}
+      />
+    );
+    expect(screen.queryByTestId("project-quick-view")).not.toBeInTheDocument();
+  });
+
+  it("calls setShowCartNotification when close button is clicked", () => {
+    const mockSetShow = vi.fn();
+    render(
+      <CartNotification
+        showCartNotification={true}
+        setShowCartNotification={mockSetShow}
+        currentProjectAddedToCart={mockProject}
+      />
+    );
+
+    fireEvent.click(screen.getByText("Close"));
+    expect(mockSetShow).toHaveBeenCalledWith(false);
+  });
+
+  it('opens cart in a new tab when "View my cart" button is clicked', () => {
+    // Mocking window.open
+    const mockOpen = vi.fn();
+    global.open = mockOpen;
```

Comment:
Ouch. Not great but I get why it's needed. Can you set it as a mock in a `beforeEach` and make sure it's reset between tests?

Reply:
Yes, will fix.